### PR TITLE
Simple Notepad in Dashboard

### PR DESCRIPTION
**Pull request by F3PiX**
Simple notepad added, and adjustments in UX.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -141,6 +141,10 @@ def current_student?
       student.any?
   end
 
+  def notepad
```
Comment (carpodaster) [inline app/models/user.rb]: I'm not sure if I like adding `User#notepad` when it only applies to supervisor-users. I would love User have no knowledge of `Note` whatsoever. Could you move the `find_or_create` to the `Note` class and update the Supervisor::NotesController#update accordingly?

File: app/controllers/supervisor/notes_controller.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+class Supervisor::NotesController < Supervisor::BaseController
+
+  def update
+    notepad = @current_user.notepad
+    notepad.update!(notepad_params)L
```
Comment (carpodaster) [inline app/controllers/supervisor/notes_controller.rb]: The "L" shouldn't be there

File: app/views/supervisor/dashboard/index.html.slim [hunk 3]
```diff
@@ -1,30 +1,43 @@
...
+                  .performance-icon = render 'orange'
+                td
+                  '= Needs Tender Loving Care
+              tr
+                td
+                  .performance-icon = render 'red'
+                td
+                  '= Red alert
+          div.col-sm-6 class=("col-lg-4 padding" if supervised_teams.count > 2)
+            = simple_form_for @current_user.notepad, url: supervisor_notes_path do |f|
```
Comment (carpodaster) [inline app/views/supervisor/dashboard/index.html.slim]: It's better to access the `@current_user` w/o the @, ie. via its corresponding helper method.

File: app/controllers/supervisor/notes_controller.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
+class Supervisor::NotesController < Supervisor::BaseController
+
+  def update
+    notepad = @current_user.notepad
```
Comment (carpodaster) [inline app/controllers/supervisor/notes_controller.rb]: It's better to access the `@current_user` w/o the @, ie. via its corresponding helper method.

File: config/routes.rb [hunk 5]
```diff
@@ -80,9 +80,11 @@
   end
 
   get 'supervisor', to: 'supervisor/dashboard#index'
+  patch 'supervisor/notes', to: 'supervisor/notes#update'
   namespace :supervisor do
     get 'dashboard', to: 'dashboard#index'
     resources :comments, only: [:index, :create]
+    resources :notes
```
Comment (carpodaster) [inline config/routes.rb]: is this needed with the `patch` statement in place 4 lines above?

File: db/migrate/20150928180532_add_user_id_to_notes.rb [hunk 6]
```diff
@@ -0,0 +1,5 @@
+class AddUserIdToNotes < ActiveRecord::Migration
+  def change
+    add_column :notes, :user_id, :integer
```
Comment (carpodaster) [inline db/migrate/20150928180532_add_user_id_to_notes.rb]: The foreign key should also have an index. Admittedly, the table won't go Big Data anytime soon, but it's good to make a habit out of it.

**Comment by carpodaster** [comment]
Hi! I've made a few comments. Could you also add one or two screenies to illustrate what it looks like?

**Comment by roosdebildt** [comment]
Screenshot (sorry I forget that all the time!):

screen shot 2015-09-28 at 18 10 22

**Comment by F3PiX** [comment]
@carpodaster

**Comment by F3PiX** [comment]
Aaaaaand back to you @carpodaster.

**Comment by carpodaster** [comment]
:tada:

**Pull request merged**